app/src/main/java/com/example/traders/presentation/watchlist/WatchListRepository.kt

```diff
@@ -1,166 +1,167 @@
-//        }
-//    }
-//
-//    //    I will have to start and pause collecting state
-//    // It collects message emitted from websocket sharedFlow and updates list item by reassigning BinanceDataItem to new value
-//    suspend fun startCollectingBinanceTickerData() {
-//        webSocketClient.state.collect { tickerData ->
-//            val indexOfCryptoDataToUpdate = _binanceCryptoList.value.indexOfFirst {
-//                it.symbol == tickerData.symbol
-//            }
-//
-//            if (indexOfCryptoDataToUpdate != -1) {
-//                _binanceCryptoList.value = _binanceCryptoList.value.let {
-//                    val updatedList = it.toMutableList()
-//                    val itemToUpdate =
-//                        tickerData.toBinanceDataItem(updatedList[indexOfCryptoDataToUpdate].isFavourite)
-//                    if (itemToUpdate != null) {
-//                        updatedList[indexOfCryptoDataToUpdate] = itemToUpdate
-//                    }
-//                    updatedList
-//                }
-//            }
-//        }
-//    }
-//
-//    // Converts PriceTickerData tp Binance24DataItem
-//    private fun PriceTickerData?.toBinanceDataItem(isFavourite: Boolean): BinanceDataItem? {
-//        if (this == null) return null
-//        return BinanceDataItem(
-//            symbol = symbol.replace("USDT", ""),
-//            last = last,
-//            high = high,
-//            low = low,
-//            open = open,
-//            priceChange = priceChange,
-//            priceChangePercent = priceChangePercent,
-//            isFavourite = isFavourite
-//        )
-//    }
-//}
\ No newline at end of file
+
+import com.example.traders.database.CryptoDatabaseDao
+import com.example.traders.database.FavouriteCrypto
+import com.example.traders.database.PreferancesManager
+import com.example.traders.database.SortOrder
+import com.example.traders.network.BinanceApi
+import com.example.traders.network.repository.enumContains
+import com.example.traders.utils.MappingUtils.enumConstantNames
+import com.example.traders.database.FixedCryptoList
+import com.example.traders.network.models.binance24HourData.Binance24DataItem
+import com.example.traders.network.models.binance24HourData.BinanceDataItem
+import com.example.traders.network.models.binance24hTickerData.PriceTickerData
+import com.example.traders.network.webSocket.BinanceWSClient
+import kotlinx.coroutines.flow.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class WatchListRepository @Inject constructor(
+    private val web_socket_client: BinanceWSClient,
+    private val binanceApi: BinanceApi,
+    private val cryptoDao: CryptoDatabaseDao,
+    private val preferencesManager: PreferancesManager
+) {
+// Sorting is not applied here since list should only be sorted only when sort type changes
+    private val _binanceCryptoList = MutableStateFlow<List<BinanceDataItem>>(emptyList())
+    val binanceCryptoList = _binanceCryptoList
```

Comment:
Although the length of variables does not make huge impact as long as they are clear, but in this case it could be abbreviated like this:
`    val cryptoList = _binanceCryptoList
`